[flaky test] TestMetrics

<!-- Please only use this template for submitting reports about flaky tests or jobs (pass or fail with no underlying change in code) in Kubernetes CI -->

**Which jobs are flaking**:

pull-kubernetes-bazel-test

**Which test(s) are flaking**:

k8s.io/client-go/util/workqueue

**Testgrid link**:

https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-bazel-test&include-filter-by-regex=k8s.io%2Fclient-go%2Futil%2Fworkqueue

**Reason for failure**:

```
=== RUN   TestMetrics
    metrics_test.go:263: expected 0.001, got 0
--- FAIL: TestMetrics (0.00s)
```

**Anything else we need to know**:
- https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/94535/pull-kubernetes-bazel-test/1301993935169654784


To reproduce using https://godoc.org/golang.org/x/tools/cmd/stress:

```
go test -race -c ./vendor/k8s.io/client-go/util/workqueue/
stress ./workqueue.test -test.run TestMetrics -test.v -test.count=1

953 runs so far, 0 failures
1841 runs so far, 0 failures

/var/folders/7f/9xt_73f12xlby0w362rgk0s400kjgb/T/go-stress-20200904T185708-124803662
=== RUN   TestMetrics
    metrics_test.go:263: expected 0.001, got 0
--- FAIL: TestMetrics (0.00s)
FAIL
```

--- a/staging/src/k8s.io/client-go/util/workqueue/metrics_test.go
+++ b/staging/src/k8s.io/client-go/util/workqueue/metrics_test.go
@@ -252,13 +252,17 @@ func TestMetrics(t *testing.T) {
 	// use a channel to ensure we don't look at the metric before it's
 	// been set.
 	ch := make(chan struct{}, 1)
+	longestCh := make(chan struct{}, 1)
 	mp.unfinished.notifyCh = ch
+	mp.longest.notifyCh = longestCh
 	c.Step(time.Millisecond)
 	<-ch
 	mp.unfinished.notifyCh = nil
 	if e, a := .001, mp.unfinished.gaugeValue(); e != a {
 		t.Errorf("expected %v, got %v", e, a)
 	}
+	<-longestCh
+	mp.longest.notifyCh = nil
 	if e, a := .001, mp.longest.gaugeValue(); e != a {
 		t.Errorf("expected %v, got %v", e, a)
 	}

